### PR TITLE
[issue-584]Fix delete cube failure because of some exception

### PR DESCRIPTION
**Pull request by Jay357089**
issue #584

### CODE REVIEW

**Comment by Jay357089** [comment]
@ravipesala pls check.

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 1]
```diff
@@ -1190,23 +1194,37 @@ private[sql] case class CreateCube(cm: tableModel) extends RunnableCommand {
...
+          var dataPath: String = null
+          try {
+            relation = CarbonEnv.getInstance(sqlContext).carbonCatalog
+              .lookupRelation2(Seq(schemaName, cubeName))(sqlContext).asInstanceOf[CarbonRelation]
+            partitionCount = relation.cubeMeta.partitioner.partitionCount
+            dataPath = relation.cubeMeta.dataPath
+          } catch {
+            case _ =>
+              val tableIdentifier = new CarbonTableIdentifier(cm.schemaName, cm.cubeName)
+              carbonTable = org.carbondata.core.carbon.metadata.CarbonMetadata.getInstance
```
Comment (Zhangshunyu) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: The class "org.carbondata.core.carbon.metadata.CarbonMetadata" has been imported, why not using CarbonMetadata directly?

Comment (Jay357089) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: @Zhangshunyu thanks

**Comment by Jay357089** [comment]
@QiangCai pls check.

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 2]
```diff
@@ -1177,6 +1177,10 @@ private[sql] case class CreateCube(cm: tableModel) extends RunnableCommand {
           s"Table [$cubeName] already exists under database [$dbName]")
         sys.error(s"Table [$cubeName] already exists under database [$dbName]")
       }
+      else {
+        LOGGER.audit(
+          s"Table [$cubeName] already exists under database [$dbName].")
+      }
```
Comment (QiangCai) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: don't need  this else block

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/createtable/TestCubeDeletedAfterCreateFailure.scala [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.carbondata.spark.testsuite.createtable
```
Comment (QiangCai) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/createtable/TestCubeDeletedAfterCreateFailure.scala]: please move to package org.apache.spark.sql

**Comment by Jay357089** [comment]
no need to handle. close it.

**Pull request closed without merging**